aptos-move/move-examples/groth16_example/sources/groth16.move

```diff
@@ -82,6 +82,204 @@ module groth16_example::groth16 {
     use aptos_std::crypto_algebra::{deserialize, enable_cryptography_algebra_natives};
     #[test_only]
     use aptos_std::bls12381_algebra::{Fr, FormatFrLsb, FormatG1Compr, FormatG2Compr, FormatFq12LscLsb, G1, G2, Gt, Fq12, FormatGt};
+    #[test_only]
+    use aptos_std::bn254_algebra;
+    #[test_only]
+    use std::bcs;
+    #[test_only]
+    use std::vector;
+
+    // This test gives an example of how to take a proof, verification key, and public input in the decimal string format
+    // output by snarkjs and verify the proof.
+    // Documentation for the serialization formats used can be found in `aptos-core/aptos-move/framework/aptos-stdlib/sources/cryptography/X.move`,
+    // where X is bn254_algebra for BN254 and bls12381_algebra for BLS12_381
+    // The names are preserved from the snarkjs proof and verifier key JSON file format, with the
+    // exception of "IC", which has been renamed to `vk_gamma_abc`
+    #[test(fx = @std)]
+    fun test_verify_circom_proof(fx: signer) {
+        enable_cryptography_algebra_natives(&fx);
+        let a_x = 9291141442484249183824149917322150275993152355313319552386216014158050680949u256;
+        let a_y = 4751084799539532208179359846086616641767957505361605807745261011239799367574u256;
+        let a_bytes = bcs::to_bytes<u256>(&a_x);
```

Comment:
This should be `a_x_bytes`.

Reply:
It's mutated two lines down to be `a_bytes` - I felt this was clearer than calling it `a_x_bytes`, then mutating it to be the full bytes of `a`